itest/lnd_psbt_test.go

```diff
@@ -1032,6 +1039,324 @@ func runFundAndSignPsbt(ht *lntest.HarnessTest, alice *node.HarnessNode) {
+	require.NoError(t, err)
+
+	pkScript, err := txscript.PayToAddrScript(parsed)
+	require.NoError(t, err)
+
+	return pkScript
+}
+
+// getAddressBip32Derivation returns the PSBT BIP-0032 derivation info of an
+// address.
+func getAddressBip32Derivation(t testing.TB, addr string,
+	node *node.HarnessNode) (*psbt.Bip32Derivation,
+	*psbt.TaprootBip32Derivation) {
+
+	derivationPath := getAddressDerivationPath(t, addr, node)
+
+	// Extract the human-readable version of the derivation path.
+	accountPath := fmt.Sprintf(`m/%d'/%d'/%d'`, derivationPath[0],
+		derivationPath[1], derivationPath[2])
+
+	// Now we can apply the hardening offset that's required for the
+	// derivation path in the PSBT.
+	derivationPath[0] += hdkeychain.HardenedKeyStart
+	derivationPath[1] += hdkeychain.HardenedKeyStart
+	derivationPath[2] += hdkeychain.HardenedKeyStart
+
+	p2TrPurpose := waddrmgr.KeyScopeBIP0086.Purpose +
+		hdkeychain.HardenedKeyStart
+
+	accounts := node.RPC.ListAccounts(&walletrpc.ListAccountsRequest{})
+	for _, account := range accounts.Accounts {
+		if account.DerivationPath == accountPath {
+			accountKey, err := hdkeychain.NewKeyFromString(
+				account.ExtendedPublicKey,
+			)
+			require.NoError(t, err)
+
+			changeKey, err := accountKey.Derive(derivationPath[3])
+			require.NoError(t, err)
+
+			addrKey, err := changeKey.Derive(derivationPath[4])
+			require.NoError(t, err)
+
+			addrPubKey, err := addrKey.ECPubKey()
+			require.NoError(t, err)
+
+			pubKeyBytes := addrPubKey.SerializeCompressed()
+
+			// Make sure we actually derived the correct address.
+			var derivedAddr btcutil.Address
+			derivedAddr, err = btcutil.NewAddressWitnessPubKeyHash(
+				btcutil.Hash160(pubKeyBytes), harnessNetParams,
+			)
+			require.NoError(t, err)
+
+			// Derive a BIP-0086 key if it's a P2TR address.
+			if derivationPath[0] == p2TrPurpose {
+				trKey := txscript.ComputeTaprootKeyNoScript(
+					addrPubKey,
+				)
+				derivedAddr, err = btcutil.NewAddressTaproot(
+					schnorr.SerializePubKey(trKey),
+					harnessNetParams,
+				)
+				require.NoError(t, err)
+			}
+
+			require.Equal(t, addr, derivedAddr.String())
+
+			return &psbt.Bip32Derivation{
+					PubKey:    pubKeyBytes,
+					Bip32Path: derivationPath,
+				}, &psbt.TaprootBip32Derivation{
+					XOnlyPubKey: pubKeyBytes[1:],
+					Bip32Path:   derivationPath,
+				}
+		}
+	}
+
+	t.Fatalf("Derivation path for address %s not found", addr)
+
+	return nil, nil
+}
+
+// getAddressDerivationPath looks up the wallet's derivation path of an address.
+func getAddressDerivationPath(t testing.TB, addr string,
+	node *node.HarnessNode) []uint32 {
+
+	// We can't query a single address directly, so we just query all wallet
+	// addresses.
+	addresses := node.RPC.ListAddresses(
+		&walletrpc.ListAddressesRequest{},
+	)
+	for _, account := range addresses.AccountWithAddresses {
+		for _, address := range account.Addresses {
+			if address.Address == addr {
+				path, err := lntest.ParseDerivationPath(
```

Comment:
Ah ok, I see here that usage is typically done alongside the corresponding account information. Including the full path as a field in the addr can potentially simplify usage for users/devs and also this section (no need for the account look up and match loop).

Reply:
Was able to drastically simplify this by adding the full derivation path and public key to each address.